### PR TITLE
[15.0][MIG] analytic_tag_dimension: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT37712

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@pedrobaeza could you please review this?

**Comment by cesar-tecnativa** [comment]
@chienandalu could you review this?

File: analytic_tag_dimension/models/account_analytic_tag.py [hunk 1]
```diff
@@ -50,12 +50,17 @@ def write(self, vals):
                         raise ValidationError(
                             _(
                                 "You can not set two tags from same dimension.\n"
-                                " Records {} in the model {} have {}".format(
-                                    records_to_update.ids,
-                                    model._description,
-                                    same_dimension_tags.mapped("display_name"),
-                                )
+                                " Records %(records_to_update)s in the model "
+                                "%(description)s have %(display_name)s"
```
Comment (chienandalu) [inline analytic_tag_dimension/models/account_analytic_tag.py:54]: Is this change due to translation issues?

**Review by chienandalu — COMMENTED** [review]
Maybe I'm not the most indicated to review this module. Anyway, just a little comment:

File: analytic_tag_dimension/models/account_analytic_tag.py (hunk 1 above)
Comment (cesar-tecnativa) [reply analytic_tag_dimension/models/account_analytic_tag.py:54]: yes, and pre-comit and @pedrobaeza comments

**Review by chienandalu — COMMENTED** [review]
Ok, as for the module review @carlosdauden could be more suitable than me

**Comment by cesar-tecnativa** [comment]
@carlosdauden you you please review it?

**Comment by cesar-tecnativa** [comment]
@CarlosRoca13 could you please review this?

**Review by CarlosRoca13 — APPROVED** [review]

**Comment by cesar-tecnativa** [comment]
@LudLaf Please review

File: analytic_tag_dimension/models/account_analytic_tag.py [hunk 2]
```diff
@@ -50,12 +50,17 @@ def write(self, vals):
...
                             )
+                            % {
+                                "records_to_update": records_to_update.ids,
+                                "description": model._description,
+                                "display_name": same_dimension_tags.mapped(
+                                    "display_name"
+                                ),
+                            }
                         )
+
```
Comment (pedrobaeza) [inline analytic_tag_dimension/models/account_analytic_tag.py]: Don't add empty lines inside a method

File: analytic_tag_dimension/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -2,6 +2,6 @@
 * Artem Kostyuk <a.kostyuk@mobilunity.com>
 * Jaume Planas <jaume.planas@minorisa.net>
 * `Tecnativa <https://www.tecnativa.com>`__:
-
```
Comment (pedrobaeza) [inline analytic_tag_dimension/readme/CONTRIBUTORS.rst]: Don't remove this empty line that is needed for proper RST syntax

File: analytic_tag_dimension/tests/test_analytic_dimension.py [hunk 4]
```diff
@@ -194,4 +198,78 @@ def test_zz_dimension_rename(self):
...
+                "quantity": vals[2],
+            }
+            for vals in expected_values_list
+        ]
+        self.assertRecordValues(reports, expected_values_dict)
+
+    def test_invoice_report_multiple_types(self):
+        self.assertInvoiceReportValues(
+            [
+                [1500.0, 30000.0, 20.0],
```
Comment (pedrobaeza) [inline analytic_tag_dimension/tests/test_analytic_dimension.py]: This result is so dependent on the existing data that will fail when other module introduces new data. Please always assert specific and controlled data.

File: analytic_tag_dimension/tests/test_analytic_dimension.py [hunk 5]
```diff
@@ -194,4 +198,78 @@ def test_zz_dimension_rename(self):
...
+                [2350.0, 11750.0, 5.0],
+                [1500.0, 7500.0, 5.0],
+                [55.0, -55.0, -1.0],
+                [541.1, -541.1, -1.0],
+                [235.0, -1175.0, -5.0],
+            ]
+        )
+
+    def test_invoice_report_multiple_types_with_context(self):
+        self.assertInvoiceReportValues_update_custom_fields(
```
Comment (pedrobaeza) [inline analytic_tag_dimension/tests/test_analytic_dimension.py]: The same as previous one.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: analytic_tag_dimension/models/account_analytic_tag.py (hunk 2 above)
Comment (LudLaf) [reply analytic_tag_dimension/models/account_analytic_tag.py]: Fixed. Empty line removed.

File: analytic_tag_dimension/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (LudLaf) [reply analytic_tag_dimension/readme/CONTRIBUTORS.rst]: Fixed. Empty line added.

**Comment by LudLaf** [comment]
> @LudLaf Please review

Issue with a couple of empty lines (one to be added, another to be removed) is now sorted.

There's another change still pending in `analytic_tag_dimension/tests/test_analytic_dimension.py` related to "specific and controlled data".

**Review by carlosdauden — APPROVED** [review]
Attend @pedrobaeza tests comments

**Comment by cesar-tecnativa** [comment]
All changes are done. I removed unused new tests and keep compatibility with the previous version.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**